lib/server.js

```diff
@@ -1,16 +1,27 @@
 import createError from 'http-errors'
 
 import Reader from './reader.js'
+import { isFileNotThereError } from './utils/errors.js'
 import { noop } from './utils/misc.js'
 
 /** @import { FastifyPluginCallback, FastifyReply } from 'fastify' */
 /** @import { Resource } from './reader.js' */
 
 /**
- * @typedef {object} PluginOptions
+ * @typedef {object} PluginOptionsBase
  * @property {string} [prefix]
+ */
+/**
+ * @typedef {object} PluginOptionsFilepath
  * @property {string} filepath Path to styled map package (`.smp`) file
  */
+/**
+ * @typedef {object} PluginOptionsReader
+ * @property {Pick<Reader, keyof Reader>} reader SMP Reader interface (also supports ReaderWatch)
+ */
+/**
+ * @typedef {(PluginOptionsBase & PluginOptionsFilepath) | (PluginOptionsBase & PluginOptionsReader)} PluginOptions
+ */
```

Comment:
Not sure if `PluginOptionsBase` is necessary at all for this because the prefix is accessed via `fastify.prefix` in the plugin implementation, not `opts`.

Think you can remove that type def and pare this down a bit

```suggestion
/**
 * @typedef {PluginOptionsFilepath | PluginOptionsReader} PluginOptions
 */
```

Reply:
This is so when you use the plugin then you can pass prefix without a type error.

Reply:
This is so when you use the plugin then you can pass prefix without a type error.

Reply:
That's not necessary to do because `fastify.register()` wraps the plugin's options internally with the extra opts e.g. `prefix`.

Reply:
https://github.com/fastify/fastify/blob/v4.28.1/types/register.d.ts#L7-L13

Reply:
addressed via 4022ec29987ff78365dc49b22c6ff5b8ce10652d